File: src/IntroToKotlin/JavaPart/TextAdventure.java
package IntroToKotlin.JavaPart;

import java.util.Map;
import java.util.Scanner;

public class TextAdventure {
    private static Map<Integer, Location> locations = Locations.locations;

    public static void main (String[] args) {
        Scanner scanner = new Scanner(System.in);
        Integer loc = 64;

        while (true) {
            Location location = locations.get(loc);
            System.out.println(location.getDescription());

            if (loc==0)
                break;

            Map<String, Integer> exits = location.getExits();

            StringBuilder sb = new StringBuilder();
            sb.append("Available exits are ");

            for (String exit: exits.keySet())
                sb.append(exit + ", ");

            System.out.println(sb.toString());

            String direction = scanner.nextLine().toUpperCase();

            if (exits.containsKey(direction))
                loc = exits.get(direction);
            else
                System.out.println("\n----You cannot go there----\n");
        }
    }
}

File: src/IntroToKotlin/KotlinPart/TextAdventure.kt
package IntroToKotlin.KotlinPart

// NO SEMICOLONS!!
fun main(args: Array<String>) {
    // VAL -> CONSTANTES
    // VAR -> VARIABLES
    val locations = readLocationInfo()
    var loc = 64

    while (true) {
        // ?: IF THIS IS NULL THEN ASSIGN X
        val location = locations[loc] ?: Location(0, "Sorry, something went wrong, Terminating the game")

        println(location.description)

        if (location.locationId == 0) {
            break
        }

        print("Available exits are ")
        // FOE EACH DE UN MAP
        location.exits.keys.forEach {
            // TIENE TEMPLATES COMO JS!!!
            print("$it, ")
        }

        // PROTEGEMOS CONTRA NULL con ?.
        // SI NULL ES Z ?:
        val direction = readLine()?.toUpperCase() ?: "Z"
        if (location.exits.containsKey(direction))
            // !! ABSOLUTELY SURE THIS IS NOT NULL
            loc = location.exits[direction]!!
        else
            println("You can't go in that direction")
    }
}